### PR TITLE
Add a warning message when one created collector needs extra RBAC permissions and the service account doesn't have them

### PR DESCRIPTION
**Pull request by iblancasa**
- Resolves: #3432

### CODE REVIEW

**Review by swiatekm — COMMENTED** [review]
I'm not confident we aren't missing some edge cases here. This PR modifies some core components, but doesn't include any new tests. Maybe we can include some test using envtest to check that the webhook behaves the way we expect?

File: internal/manifests/collector/collector.go [hunk 1]
```diff
@@ -80,6 +83,28 @@ func Build(params manifests.Params) ([]client.Object, error) {
...
+			sa, err := manifests.Factory(ServiceAccount)(params)
+			if err != nil {
+				return nil, err
+			}
+			saName = sa.GetName()
+		}
+		warnings, err := CheckRbacRules(params, saName)
+		if err != nil {
+			return nil, fmt.Errorf("error checking RBAC rules for serviceAccount %s: %w", saName, err)
+		}
```
[the remark this answers is not in the document]
Comment (iblancasa) [reply internal/manifests/collector/collector.go:99]: Good point. What do you think about checking only if the user provided the Service Account and show the warning in the other case?

**Comment by iblancasa** [comment]
> I'm not confident we aren't missing some edge cases here. This PR modifies some core components, but doesn't include any new tests. Maybe we can include some test using envtest to check that the webhook behaves the way we expect?

I will check how to do that 👍🏻

File: internal/manifests/collector/collector.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (swiatekm) [reply internal/manifests/collector/collector.go:99]: We shouldn't show the warning unless we're confident it's correct. For all we know, the user prepared a role and binding for the default ServiceAccount created by the operator. Ideally, we'd do the check and ignore the error if the ServiceAccount doesn't exist, though I don't know how practical that is. If it's not, then we should only do the check for user-set SAs.

[the remark this answers is not in the document]
Comment (iblancasa) [reply internal/manifests/collector/collector.go:99]: If we don't want to show the warning in those cases where the SA doesn't exist yet, it makes sense to me to not even check for the permissions. I'll work on it.

Thanks!

**Review by swiatekm — APPROVED** [review]
LGTM

**Review by jaronoff97 — APPROVED** [review]
